### ISSUE TITLE
Not able to publish to My Podcast

### ISSUE THREAD

**Issue by codenift** [LibreTime/libretime#836]
Why am I not able to publish to My Podcast? Also, publishing to Soundcloud is not working for me either. It seems like I have everything working in Libretime except My Podcast. I'm able to import an external podcas, but if I want to publish a track from the "Tracks" page, the Publish button doesn't do anything besides giving an error:

Failed to load resource: the server responded with a status of 500 (Internal Server Error)
http://--ip-address--/rest/media/883/publish

And for the record:  angular.min.js.map seems to be missing.
Failed to load resource: the server responded with a status of 404 (Not Found)

**Comment by Robbt** [comment]
Publish to soundcloud does not work, this feature was never fully tested or implemented as far as I know.
The podcast functionality is also limited and needs more work. It was a feature that wasn't fully implemented in the codebase we forked from airtime and some work has been done on making work it evidently is not working in all cases. 

If you want to help implement a fix we could look for a specific error in your zend-php.log in /var/log/airtime

**Comment by codenift** [comment]
Ok Thanks. I just started last week learning Linux (which I never saw myself jumping into), so I'm still new to this. But since I like this project and need it for two internet radio stations I have using Samcloud, I want to switch to this. I do have a background with PHP among other languages and would like to contribute once I study how everything works here because I see the application files from my home/ubuntu/libretime/airtime_mvc passed on to /usr/share/airtime/php/airtime_mvc/ which I'm not able to edit unless I 'sudo nano' from terminal. A bit different from what I'm used to. At this point, I haven't even been able to log in into Ubuntu (w/SSH) using Atom so I won't be able to debug and find the problem easily. Are these files being generated to that folder? How do I go about editing that file and testing it. Say I want to debug PodcastEpisodeService.php, what's the best way to do it? 

This is what I get from zend-php.log right after clicking on Publish:

`2019-07-04T03:59:11+00:00 ERR (3): 64.111.98.163 [Logging.php:166 - loggingShutdownCallback()] - Uncaught ArgumentCountError: Too few arguments to function Application_Service_PodcastEpisodeService::_buildEpisode(), 4 passed in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php on line 236 and exactly 6 expected in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114
Stack trace:
#0 /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php(236): Application_Service_PodcastEpisodeService->_buildEpisode('1', 'http://64.111.9...', 'http://64.111.9...', 'Thu, 04 Jul 201...')
#1 /usr/share/airtime/php/airtime_mvc/application/services/PublishService.php(23): Application_Service_PodcastEpisodeService->publish('883')
#2 /usr/share/airtime/php/airtime_mvc/application/modules/rest/controllers/MediaController.php(223): Application_Service_PublishService::publish('883', Array)
#3 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Action.php(516): Rest_MediaController->publishAction()
#4 /usr/share/airtime/ at /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114
array (
  'type' => 1,
  'message' => 'Uncaught ArgumentCountError: Too few arguments to function Application_Service_PodcastEpisodeService::_buildEpisode(), 4 passed in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php on line 236 and exactly 6 expected in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114
Stack trace:
#0 /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php(236): Application_Service_PodcastEpisodeService->_buildEpisode(\'1\', \'http://64.111.9...\', \'http://64.111.9...\', \'Thu, 04 Jul 201...\')
#1 /usr/share/airtime/php/airtime_mvc/application/services/PublishService.php(23): Application_Service_PodcastEpisodeService->publish(\'883\')
#2 /usr/share/airtime/php/airtime_mvc/application/modules/rest/controllers/MediaController.php(223): Application_Service_PublishService::publish(\'883\', Array)
#3 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Action.php(516): Rest_MediaController->publishAction()
#4 /usr/share/airtime/',
  'file' => '/usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php',
  'line' => 114,
)`

**Comment by codenift** [comment]
Ok Thanks. I just started last week learning Linux (which I never saw myself jumping into), so I'm still new to this. But since I like this project and need it for two internet radio stations I have using Samcloud, I want to switch to this. I do have a background with PHP among other languages and would like to contribute once I study how everything works here because I see the application files from my home/ubuntu/libretime/airtime_mvc passed on to /usr/share/airtime/php/airtime_mvc/ which I'm not able to edit unless I 'sudo nano' from terminal. A bit different from what I'm used to. At this point, I haven't even been able to log in into Ubuntu (w/SSH) using Atom so I won't be able to debug and find the problem easily. Are these files being generated to that folder? How do I go about editing that file and testing it. Say I want to debug PodcastEpisodeService.php, what's the best way to do it?

This is what I get from zend-php.log right after clicking on Publish:

2019-07-04T03:59:11+00:00 ERR (3): -my-ip-address- [Logging.php:166 - loggingShutdownCallback()] - Uncaught ArgumentCountError: Too few arguments to function Application_Service_PodcastEpisodeService::_buildEpisode(), 4 passed in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php on line 236 and exactly 6 expected in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114 

Stack trace:
 #0 /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php(236): Application_Service_PodcastEpisodeService->_buildEpisode('1', 'http://64.111.9...', 'http://64.111.9...', 'Thu, 04 Jul 201...') 

#1 /usr/share/airtime/php/airtime_mvc/application/services/PublishService.php(23): Application_Service_PodcastEpisodeService->publish('883') 

#2 /usr/share/airtime/php/airtime_mvc/application/modules/rest/controllers/MediaController.php(223): Application_Service_PublishService::publish('883', Array) 

#3 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Action.php(516): Rest_MediaController->publishAction() 

#4 /usr/share/airtime/ at /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114 

array ( 
       'type' => 1, 
       'message' => 'Uncaught ArgumentCountError: Too few arguments to function Application_Service_PodcastEpisodeService::_buildEpisode(), 4 passed in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php on line 236 and exactly 6 expected in /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php:114 

Stack trace: 
#0 /usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php(236): Application_Service_PodcastEpisodeService->_buildEpisode(\'1\', \'http://64.111.9...\', \'http://64.111.9...\', \'Thu, 04 Jul 201...\') 

#1 /usr/share/airtime/php/airtime_mvc/application/services/PublishService.php(23): Application_Service_PodcastEpisodeService->publish(\'883\') 

#2 /usr/share/airtime/php/airtime_mvc/application/modules/rest/controllers/MediaController.php(223): Application_Service_PublishService::publish(\'883\', Array) 

#3 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Action.php(516): Rest_MediaController->publishAction() 

#4 /usr/share/airtime/', 'file' => '/usr/share/airtime/php/airtime_mvc/application/services/PodcastEpisodeService.php', 'line' => 114, )

**Comment by Robbt** [comment]
Yeah LibreTime is unforunately not the simplest platform to develop on because of the framework it depends upon currently.

So for studying the code you are correct that the files are copied and so you can find all of the files from your install in the libretime/airtime_mvc folder.

As far as development goes and troubleshooting, it is possible to do this on your live server but I would suggest you might benefit from setting up a separate vagrant virtualbox environment for development and testing - see http://libretime.org/vagrant/ - this will provide you with a standalone server running on your computer that you can edit the files directly and they will automatically change. It maps the local files to the server files and you can access it via http://localhost:8080 - installing virtualbox and vagrant can be a chore if you have never done it before but it is worth it if you want to try to contribute a fix or even diagnose it further.

My development time is severely limited due to

**Comment by Robbt** [comment]
So the code being ran is line 236 on  **airtime_mvc/application/services/PodcastEpisodeService.php** 
which is running into the issue with _build_episode -```

     * Publish the file with the given file ID to the station podcast
     *
     * @param int $fileId ID of the file to be published
     */
    public function publish($fileId) {
        $id = Application_Model_Preference::getStationPodcastId();
        $url = $guid = Application_Common_HTTPHelper::getStationUrl()."rest/media/$fileId/download";
        if (!PodcastEpisodesQuery::create()
            ->filterByDbPodcastId($id)
            ->findOneByDbFileId($fileId)) {  // Don't allow duplicate episodes
            $e = $this->_buildEpisode($id, $url, $guid, date('r'));
            $e->setDbFileId($fileId)->save();
        }
    }

```
which is defined in 
PodcastEpisodeService.php:    private function _buildEpisode($podcastId, $url, $guid, $publicationDate, $title, $description) {

and I see the problem. It now expects a $title and $description, which I think we added to make the downloadable podcasts have this, but we failed to realize it would impact the built-in episodes. I'll see if I can put together a quick fix.

**Comment by Robbt** [comment]
So you can try the fix I put together in the pull request above and see if it fixes this. I'll try to test it later but this might work as a very simple workaround fix.

**Comment by codenift** [comment]
Great! Thank you so much. Will definitely look into it.

**Comment by korotkov13** [comment]
Hi guys! 
I have the same issue, and the fix #837 didn't solve it for me. I got this error in zendphp.log


```
[MediaController.php:228 - publishAction()] - Unable to execute INSERT statement [INSERT INTO "podcast_episodes" ("id", "podcast_id", "publication_date"$
DETAIL:  Failing row contains (28, null, 1, 2019-08-11 21:55:14, https://myradio/rest/media/4/download, https://myradio/rest/media/4/download, null, null).]


```
I Also tried to use an old PodcastEpisodeService.php from my working version, but it also 
have the same error in zendphp.log
Any ideas where do dig?

**Comment by Robbt** [comment]
Your log looks truncated, I'm not sure why the podcast ID would be null - that might be a place to look also see what it says in Application_Service_PublishService::publish($id, $data) - as that appears to be where the failing insert code is. This seems like a slightly different error.

**Comment by farggus** [comment]
Hi everyone!
I have the same issue, and the fix [#837](https://github.com/LibreTime/libretime/pull/837) didn't solve it for me. 
I got this error in /var/log/airtime/zendphp.log, the same as @korotkov13 

> 2020-03-08T23:08:16+00:00 ERR (3): -my-ip-address [MediaController.php:228 - publishAction()] - Unable to execute INSERT statement [INSERT INTO "podcast_episodes" ("id", "podcast_id", "publication_date", "download_url", "episode_guid") VALUES (:p0, :p1, :p2, :p3, :p4)] [wrapped: SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "episode_title" violates not-null constraint
DETAIL:  Failing row contains (47, null, 1, 2020-03-08 23:08:16, http://myradio/rest/media/7/download, http://myradio/rest/media/7/download, null, null).]

Which way should I look in order to try to fix it?
